### ISSUE TITLE
Simulate small T, no noise, with Norris labels

### ISSUE THREAD

**Issue by MatthewJA** [chengsoonong/crowdastro#133]

**Comment by MatthewJA** [comment]
60% accuracy, same as with the top 10 annotators in #126. Testing on the test set:

![image](https://cloud.githubusercontent.com/assets/2681312/17541726/955e635c-5f05-11e6-8d7d-2218cb7a65ca.png)

Testing on the train set:

![image](https://cloud.githubusercontent.com/assets/2681312/17541727/9a0ec36a-5f05-11e6-9d6b-2acf111a477e.png)

**Comment by MatthewJA** [comment]
Worth noting: This is worse than majority vote.

**Comment by chengsoonong** [comment]
I am guessing that you are using predict_proba, which introduces a complicated nonlinearity.

Please use the score for the plots.
http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.decision_function

For training, you can also report the performance of the noise free data (i.e. directly use Norris labels). This gives you a baseline to compare against.

Based on the shape of your curves above, I am guessing that your features are not good enough to identify a portion of the positives. No concrete suggestion on how to fix, so leave this for later.

**Comment by MatthewJA** [comment]
> I am guessing that you are using predict_proba, which introduces a complicated nonlinearity.

Ah, yeah, I am. I'll fix that.

> Please use the score for the plots.

Is this just ax + b? I'm not using `sklearn` so I can't use `decision_function` but ax + b is easy enough.

> For training, you can also report the performance of the noise free data (i.e. directly use Norris labels). This gives you a baseline to compare against.

You mean without partial observation of labels? I've done partial observation here with no noise, five annotators, and each annotator (correctly) labelling 50% of all objects at random.

> Based on the shape of your curves above, I am guessing that your features are not good enough to identify a portion of the positives. No concrete suggestion on how to fix, so leave this for later.

That makes sense to me. I suspect that better CNN features could help, so I am interested to see if I can get good autoencoded features.

**Comment by MatthewJA** [comment]
Test set:
![image](https://cloud.githubusercontent.com/assets/2681312/17543877/00bbe24e-5f17-11e6-862f-553fd9967d0a.png)

Train set:
![image](https://cloud.githubusercontent.com/assets/2681312/17543892/18205a64-5f17-11e6-8e4d-28be36ea34e5.png)

**Comment by chengsoonong** [comment]
> > For training, you can also report the performance of the noise free data (i.e. directly use Norris  labels). This gives you a baseline to compare against.
> 
> You mean without partial observation of labels? I've done partial observation here with no noise,  five annotators, and each annotator (correctly) labelling 50% of all objects at random.

I mean just standard supervised learning, fully observed, one example, one label setting.

**Comment by MatthewJA** [comment]
Ah, right.

Test set:
![image](https://cloud.githubusercontent.com/assets/2681312/17544052/3d2cb68a-5f18-11e6-8494-8a2e7dc15c9b.png)

Train set:
![image](https://cloud.githubusercontent.com/assets/2681312/17544059/4d3ca580-5f18-11e6-9167-4b01b2bbe583.png)

72% accuracy (so matches #136) with a very similar distribution.

**Comment by MatthewJA** [comment]
Looks like I might want to see if I can get some better features.

#99

**Comment by MatthewJA** [comment]
I now get around 68% accuracy with my revised code, c.f. 69% accuracy with the Norris groundtruth and `sklearn`.

Going to try upsampling instead of downsampling.

**Comment by MatthewJA** [comment]
Upsampling gives 25% accuracy, haha. Maybe I should modify the EM stuff to accommodate for class weights? I'm not really sure how though.
